pkg/controller/apimanagerbackup/apimanagerbackup_logic_reconciler.go

```diff
@@ -0,0 +1,398 @@
+		return result, err
+	}
+
+	result, err = r.reconcileBackupCompletion()
+	if result.Requeue || err != nil {
+		return result, err
+	}
+
+	return result, err
+}
+
+func (r *APIManagerBackupLogicReconciler) reconcileBackupInS3Destination() (reconcile.Result, error) {
+	// TODO implement
+	return reconcile.Result{}, nil
+}
+
+func (r *APIManagerBackupLogicReconciler) reconcileBackupInPVCDestination() (reconcile.Result, error) {
+	var res reconcile.Result
+	var err error
+
+	err = r.reconcileBackupDestinationPVC()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	res, err = r.reconcileBackupDestinationPVCStatus()
+	if res.Requeue || err != nil {
+		return res, err
+	}
+
+	res, err = r.reconcileBackupSecretsAndConfigMapsToPVCJob()
+	if res.Requeue || err != nil {
+		return res, err
+	}
+
+	res, err = r.reconcileAPIManagerCustomResourceBackupToPVCJob()
+	if res.Requeue || err != nil {
+		return res, err
+	}
+
+	res, err = r.reconcileBackupSystemFileStoragePVCToPVCJob()
+	if res.Requeue || err != nil {
+		return res, err
+	}
+
+	return res, err
+}
+
+func (r *APIManagerBackupLogicReconciler) reconcileBackupDestinationPVC() error {
+	desired := r.apiManagerBackup.BackupDestinationPVC()
+	if desired == nil {
+		return nil
+	}
+
+	// TODO create mutator function for PVC ?
+	err := r.ReconcileResource(&v1.PersistentVolumeClaim{}, desired, reconcilers.CreateOnlyMutator)
+
+	return err
+}
+
+func (r *APIManagerBackupLogicReconciler) reconcileBackupSecretsAndConfigMapsToPVCJob() (reconcile.Result, error) {
+	desired := r.apiManagerBackup.BackupSecretsAndConfigMapsToPVCJob()
+	if desired == nil {
+		return reconcile.Result{}, nil
+	}
+
+	// TODO create Mutator function for Jobs??
+	existing := &batchv1.Job{}
+	// Check if this step has finished
+	if r.cr.SecretsAndConfigMapsBackupStepFinished() {
+		return reconcile.Result{}, nil
+	}
+
+	// Check if backup substep has completed
+	if !r.cr.SecretsAndConfigMapsBackupSubStepFinished() {
+		err := r.ReconcileResource(existing, desired, reconcilers.CreateOnlyMutator)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		substepFinished := true
+		if existing.Status.Succeeded != *desired.Spec.Completions {
+			r.Logger().Info("Job has still not finished", "Job Name", desired.Name, "Actively running Pods", existing.Status.Active, "Failed pods", existing.Status.Failed)
+			return reconcile.Result{Requeue: true}, nil
+		}
+
+		r.cr.Status.SecretsAndConfigMapsBackupSubStepFinished = &substepFinished
+		err = r.UpdateResourceStatus(r.cr)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		r.Logger().Info("Job finished successfully. Requeing", "Job Name", desired.Name)
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	// Check if cleanup substep has completed
+	if !r.cr.SecretsAndConfigMapsCleanupSubStepFinished() {
+		common.TagToObjectDeleteWithPropagationPolicy(desired, metav1.DeletePropagationForeground)
+		err := r.ReconcileResource(existing, desired, reconcilers.CreateOnlyMutator)
```

Comment:
if `r.cr.SecretsAndConfigMapsCleanupSubStepFinished()` is false, you can assume Job object has been created. can't you?

Why not like this?
```
err = r.GetResource(common.ObjectKey(desired), existing)      
if err != nil 
    if !errors.IsNotFound(err) {                    
    return reconcile.Result{}, err                 
    }
    // Job Not found
    // Here check config maps cleanup done
       r.cr.Status.SecretsAndConfigMapsCleanupSubStepFinished = &substepFinished
     ...
}                                                             


```

Reply:
I'm not sure I'm following you.
If r.cr.SecretsAndConfigMapsCleanupSubStepFinished() is false it means the Job object has been created and finished correctly indeed.

In this step you have to delete the Job in case it still exists so you have to end up calling the ReconcileResource object anyway with the deletion tag, and you can only mark as finished when it's fully deleted. You cannot call the ReconcileResource to delete and then unconditionally continue. I tried to explain that in a TODO just below. With the code you shown me it still lacks the deletion part, which has to be added anyway.

If you look a little bit below you'll see that I might be doing sort of what you say, but after performing the deletion:
```
		err = r.GetResource(common.ObjectKey(desired), existing)
		if err != nil && !errors.IsNotFound(err) {
			return reconcile.Result{}, err
		}
```

Reply:
EDIT: I've added more code related to feedback you left me in https://github.com/3scale/3scale-operator/pull/358#discussion_r425740366. Take that into account for this discussion also

Reply:
This is obsolete as the state check was refactored following another approach. Marking as resolved.